File: SEM 3/OOPL/LAB3/q1.java
import java.util.Scanner;
import java.lang.Math;
class Complex
{
    int a;
    int b;
    void set(int a,int b)
    {
        this.a=a;
        this.b=b;
    }
    public static void sum(Complex val1,Complex val2,Complex sum)
    {   
        sum.a = val1.a + val2.a;
        sum.b = val1.b + val2.b;
    }
    public static void diff(Complex val1,Complex val2,Complex diff)
    {
        diff.a = val1.a - val2.a;
        diff.b = val1.b - val2.b;
    }
    public static void Display(Complex temp)
    {
        String c="+";
        if(temp.b<0)
        {
            c="-";
        }
        int t=Math.abs(temp.b);
        System.out.println(temp.a+c+t+"i");
    }
}
class q1
{
    public static void main(String[] args)
    {
        int a,b,c,d;
        Scanner sc=new Scanner(System.in);
        Complex val1 = new Complex();
        Complex val2 = new Complex();
        Complex sum = new Complex();
        Complex diff = new Complex();
        System.out.println("Enter Real and Imaginary Parts of Complex Number 1 ");
        a=sc.nextInt();
        b=sc.nextInt();
        System.out.println("Enter Real and Imaginary Parts of Complex Number 2 ");
        c=sc.nextInt();
        d=sc.nextInt();
        val1.set(a,b);
        val2.set(c,d);
        System.out.println("First Complex Number is : ");
        val1.Display(val1);
        System.out.println("Second Complex Number is :");
        val2.Display(val2);
        System.out.println("Sum of Given Complex Numbers is :");
        Complex.sum(val1,val2,sum);
        sum.Display(sum);
        System.out.println("Difference of Given Complex Numbers is :");
        Complex.diff(val1,val2,diff);
        diff.Display(diff);
    }    
}

File: SEM 3/DSAL/LAB3/q1.c
#include <stdio.h>
#include <stdlib.h>

struct Complex
{
	int re;
	int im;
};
void set(struct Complex *ptr,int a,int b)
{
	ptr->re=a;
	ptr->im=b;
}
void display(struct Complex *ptr)
{
	char c='+';
	int t=abs(ptr->im);
	if(ptr->im < 0)
	{
		c='-';
	}
	printf("%d%c%di\n",ptr->re,c,t);

}
void add(struct Complex *add,struct Complex *h1,struct Complex *h2)
{
	add->re = h1->re + h2->re;
	add->im = h1->im + h2->im;
}
void sub(struct Complex *sub,struct Complex *h1,struct Complex *h2)
{
	sub->re = h1->re - h2->re;
	sub->im = h1->im - h2->im;
}
void mult(struct Complex *mult,struct Complex *h1,struct Complex *h2)
{
	mult->re=(h1->re*h2->re)-(h1->im*h2->im);
	mult->im=(h1->re*h2->im)+(h2->re*h1->im);
}
void main()
{
	struct Complex c1,c2,sum,diff,product;
	int m,n,p,q;
	printf("Enter real and imaginary parts of C1 : \t");
	scanf("%d %d",&m,&n);
	struct Complex *p1=&c1,*p2=&c2,*sum1=&sum,*diff1=&diff,*product1=&product;
	set(p1,m,n);
	printf("Enter real and imaginary parts of C2 : \t");
	scanf("%d %d",&p,&q);
	set(p2,p,q);
	display(p1);
	display(p2);
	add(sum1,p1,p2);
	printf("Sum is ");
	display(sum1);
	sub(diff1,p1,p2);
	printf("Difference is ");
	display(diff1);
	mult(product1,p1,p2);
	printf("Product is ");
	display(product1);
}